ingestion/src/metadata/ingestion/source/database/incremental_metadata_extraction.py

```diff
@@ -0,0 +1,145 @@
+from metadata.utils.logger import ingestion_logger
+
+logger = ingestion_logger()
+
+
+class IncrementalConfig(BaseModel):
+    """Holds the Configuration to extract the Metadata incrementally, if enabled."""
+
+    enabled: bool
+    start_timestamp: Optional[int] = None
+
+    @property
+    def start_datetime_utc(self) -> Optional[datetime]:
+        if self.start_timestamp:
+            return datetime.utcfromtimestamp(self.start_timestamp / 1000)
+        return None
+
+    @classmethod
+    def create(
+        cls,
+        incremental: Optional[bool],
+        pipeline_name: Optional[str],
+        metadata: OpenMetadata,
+    ) -> "IncrementalConfig":
+        """Returns the IncrementalConfig based on the flow defined on the INcrementalConfigCreator."""
+        return IncrementalConfigCreator(incremental, pipeline_name, metadata).create()
+
+
+class IncrementalConfigCreator:
+    """Helper class to create an IncrementalConfig instance automagically."""
+
+    def __init__(
+        self,
+        incremental: Optional[Incremental],
+        pipeline_name: Optional[str],
+        metadata: OpenMetadata,
+    ):
+        self.incremental = incremental
+        self.pipeline_name = pipeline_name
+        self.metadata = metadata
+
+    def _calculate_pipeline_status_parameters(self) -> Tuple[int, int]:
+        """Calculate the needed 'start' and 'end' parameters based on the 'lookbackDays'."""
+        now = datetime.now()
+
+        # We multiply the value by 1000 because our backend uses epoch_milliseconds instead of epoch_seconds.
+        start = int(
+            (now - timedelta(days=self.incremental.lookbackDays)).timestamp() * 1000
+        )
+        end = int(now.timestamp() * 1000)
+
+        return start, end
+
+    def _get_pipeline_statuses(self) -> Optional[List[PipelineStatus]]:
+        """Retrieve all the pipeline statuses between 'start' and 'end'."""
+        if not self.pipeline_name:
+            return None
+
+        start, end = self._calculate_pipeline_status_parameters()
+
+        return self.metadata.get_pipeline_status_between_ts(
+            self.pipeline_name, start, end
+        )
+
+    def _get_last_success_timestamp(
+        self, pipeline_statuses: List[PipelineStatus]
+    ) -> Optional[int]:
+        """Filter the pipeline statuses to get the last time the pipeline was run succesfully."""
+        return max(  # pylint: disable=R1728
+            [
+                pipeline.startDate.__root__
+                for pipeline in pipeline_statuses
+                if pipeline.pipelineState == PipelineState.success
+                and pipeline.startDate
+            ]
+        )
+
+    def _add_safety_margin(self, last_success_timestamp: int) -> int:
+        """Add some safety margin to the last successful run timestamp based on the 'safetyMarginDays'."""
+        # Hours * Minutes * Seconds * Milliseconds
+        milliseconds_in_one_day = 24 * 60 * 60 * 1000
```

Comment:
Could this be a constant in the module? or even a constant in `constants.py`

Reply:
Not strong opinions here. Since it's only used in one place I don't see the need, but if you'd rather have it we can have it as a constant in the module. Absolutely no harm (=